src/index-router/index-router.ts

```diff
@@ -47,114 +52,60 @@ export class IndexRouter extends Contract<IndexRouterContractInterface> {
   /**
    * ### Mint
    *
-   * @param options mint options
+   * @param index index address
    * @param sellAmount token's  amount
-   * @param sellToken (optional) erc20 token
-   * @param permitOptions (optional) permit options for transaction
+   * @param params (optional) quotes for erc20 token deposit
+
    *
    * @returns mint transaction
    */
   async mintSwap(
-    options: IIndexRouter.MintSwapParamsStruct,
+    index: Address,
```

Comment:
please keep whole index-router.ts unchanged, its idea is to wrap IndexRouter.sol contract, not to actually do any logic.

Instead, add instance of IndexDepositRouter to constructor params of AutoRouter and call IndexDepositRouter directly from selectBuy and buyMint methods

Feel free to argue, but in my opinion it would be much more clear and efficient in terms of memory (not to construct IndexDepositRouter inside mint execution process)